accounts/utils.py

```diff
@@ -0,0 +1,80 @@
+from datetime import timedelta
+
+import requests
+from django.utils import timezone
+
+from accounts.settings import accounts_settings
+
+
+def authenticated_request(user, method, url,
+                          params=None, data=None, headers=None, cookies=None, files=None,
+                          auth=None, timeout=None, allow_redirects=True, proxies=None,
+                          hooks=None, stream=None, verify=None, cert=None, json=None):
+    """
+    Helper method to make an authenticated request using the user's access token
+    NOTE be ABSOLUTELY sure you only make a request to Penn Labs products, otherwise
+    you will expose user's access tokens to the URL you provide and bad things will
+    happen
+    """
+
+    # Access token is expired. Try to refresh access token
+    if user.accesstoken.expires_at < timezone.now():
+        if not refresh_access_token(user):
+            return None  # Couldn't update access token
```

Comment:
Do we want error handling in this function? Maybe throwing custom exceptions?

Reply:
I'm not sure. The only time this should happen is if platform is down, or if someone manually revokes a refresh token. I don't know if it's worth handling, but we could if it makes sense.

Reply:
Fixed